stl/src/cthread.cpp

```diff
@@ -103,9 +105,45 @@ _CRTIMP2_PURE _Thrd_id_t __cdecl _Thrd_id() noexcept { // return unique id for c
 }
 
 _CRTIMP2_PURE unsigned int __cdecl _Thrd_hardware_concurrency() noexcept { // return number of processors
-    SYSTEM_INFO info;
-    GetNativeSystemInfo(&info);
-    return info.dwNumberOfProcessors;
+    // Most devices have only one processor group and thus have the same buffer_size
+#if defined(_M_X64) || defined(_M_ARM64) // 16 bytes per group
+    constexpr int stack_buffer_size = 48;
+#elif defined(_M_IX86) || defined(_M_ARM) // 12 bytes per group
+    constexpr int stack_buffer_size = 44;
+#else
+    constexpr int stack_buffer_size = 0;
+#endif
+    alignas(SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX) unsigned char buffer[stack_buffer_size];
+    using buffer_ptr_t = PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX;
+    DWORD buffer_size  = stack_buffer_size;
+
+    const auto try_query = [](buffer_ptr_t buffer_ptr, PDWORD buffer_size) _STATIC_CALL_OPERATOR noexcept {
+        unsigned int count = 0;
+        _ASSERT(buffer_ptr != nullptr);
+
+        if (GetLogicalProcessorInformationEx(RelationProcessorPackage, buffer_ptr, buffer_size) == TRUE) {

```

Comment:
We conventionally don't use `int` as conditional directly, however, for WinAPI `BOOL` we make an exception, and treat it a `BOOL` directly.

(Comparing `BOOL` to `TRUE` is terrible idea generally)